lib/puppet/type/scheduled_task.rb

```diff
@@ -155,7 +155,8 @@ def insync?(current)
           * `which_occurrence` **(Required)** --- The occurrence of the chosen weekday
             when the task should run. Must be one of `first`, `second`, `third`,
             `fourth`, or `fifth`.
-
+      * For `logon` triggers:
+          * `user_id` --- The user account that is used to run the task at logon.
```

Comment:
Is it worth specifying that this can/can't be local or AD accounts?

Reply:
Also, is it worth adding the readme example into the type docs, too?

Reply:
Yeah, I think it would be useful to be a bit more specific, like the `ILogonTrigger` docs.

https://docs.microsoft.com/en-us/windows/desktop/taskschd/logontrigger-userid#property-value

```
The identifier of the user. For example, "MyDomain\MyName" or for a local account, "Administrator".

This property can be in one of the following formats:

User name or SID: The task is started when the user logs on to the computer.
NULL: The task is started when any user logs on to the computer.

Remarks

If you want a task to be triggered when any member of a group logs on to the computer rather than when a specific user logs on, then do not assign a value to the LogonTrigger.UserId property. Instead, create a logon trigger with an empty LogonTrigger.UserId property and assign a value to the principal for the task using the Principal.GroupId property.
```

Reading those docs I don't think we tested `NULL` either.